src/Util.php

```diff
@@ -173,6 +174,46 @@ public static function removeFooter(string $payload): string
         return $payload;
     }
 
+    /**
+     * Strip all newlines (CR, LF) characters from a string.
+     *
+     * @param string $input
+     * @return string
+     */
+    public static function stripNewlines(string $input): string
+    {
+        $bytes = SodiumUtil::stringToIntArray($input);
+        $length = count($bytes);
+
+        // First value is a dummy value, to overwrite it in constant-time
+        $return = array_fill(0, $length + 1, 0);
+        // Output index:
+        $j = 1;
+
+        // Now let's strip:
+        for ($i = 0; $i < $length; ++$i) {
+            $char = ($bytes[$i]);
+
+            // Determine if we're stripping this character or not?
+            $isCR = ((($char ^ 0x0d) - 1) >> 8) & 1;
+            $isLF = ((($char ^ 0x0a) - 1) >> 8) & 1;
+            $isNewline = $isCR | $isLF;
+
+            // Set destination index: 0 if $isNewLine, $j otherwise
+            $swap = -$isNewline;
+
+            // if ($isNewLine), $dest === 0, else $dest === $j
+            $dest = (~$swap & $j) ^ $swap;
+
+            // Now let's overwrite the index (0 or $j) with $char:
+            $return[$dest] = $char;
```

Comment:
This memory access might be problematic if PHP dynamically allocates these two values in different memory pages and there's a visible timing difference between accessing the two, but there's not much we can do to prevent this granular level of leakage from a PHP script.